### PR TITLE
Importers: migrate to options class

### PR DESCRIPTION
**Pull request by Maffooch**
Moves the parsing of kwargs to a central `ImporterOptions` class where validation can occur and defaults be set

### CODE REVIEW

**Comment by Maffooch** [comment]
This is more of a restructure/reorg rather and a rewrite. The functionalities are otherwise the same. I can take a look to see if that issue linked has a quick solution

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

File: dojo/importers/base_importer.py [hunk 1]
```diff
@@ -237,58 +192,62 @@ def parse_dynamic_test_type_findings_from_tests(
...
+        self,
+        parsed_findings: List[Finding],
+        **kwargs: dict,
+    ) -> List[Finding]:
+        """
+        Processes findings in chunks within N number of processes. The
+        ASYNC_FINDING_IMPORT_CHUNK_SIZE setting will determine how many
+        findings will be processed in a given worker/process/thread
+        """
+        return self.process_findings(parsed_findings, sync=True, **kwargs)
```
Comment (cneill) [inline dojo/importers/base_importer.py:246]: Should this be `sync=False`?

```suggestion
        return self.process_findings(parsed_findings, sync=False, **kwargs)
```

**Review by cneill — COMMENTED** [review]
Just a couple questions/comments

**Review by cneill — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**